InventoryManagementSystem.py

```diff
@@ -0,0 +1,237 @@
+from pathlib import Path
+import tempfile
+import shutil
+
+import logging
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+# Inventory Management System
+
+# Create a simple inventory management system for a small store. This problem will help you practice using classes, methods, loops, conditionals, and basic data structures.
+
+# Requirements:
+# 1. Create a `Product` class with the following attributes:
+#   -[x] name (string)
+#   -[x] price (float)
+#   -[x] quantity (int)
+
+class Product:
+    def __init__(self, name: str, price: float, quantity: int):
+        if price < 0:
+            raise ValueError("Price can't be negative!")
+        if quantity < 0:
+            raise ValueError("Quantity can't be negative!")
+        self.name = name
+        self.price = price
+        self.quantity = quantity
+
+    def get_value(self) -> float:
+        return self.price * self.quantity
+
+    def __repr__(self):
+        return f'{self.name}, {self.price}, {self.quantity}'
+    
+    @staticmethod
+    def from_string(product_str: str):
+        name, price, quantity = product_str.strip().split(',')
+        return Product(name, float(price), int(quantity))
+    
+# BONUS Category class
+class Category:
+    def __init__(self, category: str):
+        self.category = category
+        pass
+
+# 2. Create an `InventoryManager` class that will manage a list of `Product` objects. It should have the following methods:
+#   -[x] `add_product(self, product)`: Adds a new product to the inventory
+#   -[x] `remove_product(self, product_name)`: Removes a product from the inventory by name
+#   -[x] `update_quantity(self, product_name, new_quantity)`: Updates the quantity of a product
+#   -[x] `get_total_inventory_value(self)`: Calculates and returns the total value of all products in the inventory
+#   -[x] `list_low_stock_products(self, threshold)`: Returns a list of products with quantity below the given threshold
+
+from typing import List, Optional
+
+class InventoryManager:
+    """Manages inventory of products with file persistence.
+    
+    Attributes:
+        filename (str): Path to inventory file
+        inventory (List[Product]): List of products in inventory
+    """
+    def __init__(self, filename="inventory.txt"):
+        self.filename = filename
+        self.inventory = self.load_inventory()
+
+    def load_inventory(self):
+        inventory = []
+        try:
+            with open(self.filename, 'r') as file:
+                for line in file:
+                    inventory.append(Product.from_string(line))
+        except FileNotFoundError:
+            print(f"{self.filename} not found.")
```

Comment:
here you would want to use the error logger, like the info thing you already did. Remove the Print